algorithms/ar-lcs/cpp/longestcommonsubsequence.cpp

```diff
@@ -0,0 +1,27 @@
+#include <iostream>
+#include <string.h>
+using namespace std;
+int max(int a, int b);
+int lcs(char *X, char *Y, int m, int n)
+{
+    if (m == 0 || n == 0)
+        return 0;
+    if (X[m - 1] == Y[n - 1])
+        return 1 + lcs(X, Y, m - 1, n - 1);
+    else
+        return max(lcs(X, Y, m, n - 1), lcs(X, Y, m - 1, n));
+}
+int max(int a, int b)
+{
+    return (a > b) ? a : b;
+}
+//main 
+int main()
+{
+    char X[] = "AGGTAB";
+    char Y[] = "GXTXAYB";
+    int m = strlen(X);
+    int n = strlen(Y);
```

Comment:
Codacy found an issue: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6418120)